src/main/todos.ts

```diff
@@ -1,22 +1,38 @@
-import { getBasecampProjectUrl, sendBasecampPostRequest, sendBasecampPutRequest } from "./basecamp";
+import { getBasecampProjectUrl, PROJECT_ID, sendBasecampPostRequest, sendBasecampPutRequest } from "./basecamp";
+import { BasecampRequestMissingError } from "./error/basecampRequestMissingError";
+import { getRoles, RoleStatus } from "./role";
 
 const TODOLISTS_PATH: string = '/todolists/';
 const TODO_PATH: string = '/todos/';
+const RECORDINGS_PATH: string = '/recordings/';
 const JSON_PATH: string = '.json';
 const TODO_JSON_PATH: string = '/todos' + JSON_PATH;
+const TRASHED_TODO_JSON_PATH: string = '/status/trashed' + JSON_PATH;
+
 export const TODOLIST_ID: string= "7865336721";
 
+const DEFAULT_TODOLIST_IDENTIFIER: TodolistIdentifier = {
+    projectId: PROJECT_ID,
+    todolistId: TODOLIST_ID
+};
+
 /**
  * Creates a todo in Basecamp
  * 
  * @param todo object to put in Basecamp
  * @param todolistIdentifier id of the todolist where the todo will be created in
  * @returns the id of the created todo. This must be saved by the caller to update this todo in the future.
  */
-export function createTodo(todo: BasecampTodoRequest, todolistIdentifier: TodolistIdentifier): string {
-    const rawTodoResponse: JsonData = sendBasecampPostRequest(getCreateTodoUrl(todolistIdentifier), todo);
-    const todoResponse: BasecampTodoResponse = rawTodoResponse as BasecampTodoResponse;
-    return todoResponse.id;
+export function createTodo(request: BasecampTodoRequest | undefined, todolistIdentifier: TodolistIdentifier): string {
+    
+    if(request == undefined) {
+        throw new BasecampRequestMissingError("Missing basecamp request!");
+    } else {
+        Logger.log(`Creating new todo: "${request.content}"...\n`)
+        const rawTodoResponse: JsonData = sendBasecampPostRequest(getCreateTodoUrl(todolistIdentifier), request);
+        const todoResponse: BasecampTodoResponse = rawTodoResponse as BasecampTodoResponse;
+        return todoResponse.id;
+    }
```

Comment:
When would a request be undefined? If for some reason it does have a possibility of being undefined, I feel like we should be checking for that before even calling this function to send the POST request

Reply:
I did run into the request being undefined at some points, but that was only because earlier in the function the code was trying to get the value for an unexpected role, but that was due to a bug that I fixed even more upstream in the logic in previous commits so we shouldn't need to worry about that. I agree, it makes more sense to check for undefined requests if we are going to check for them before running the createTodo function.